libs/interpret-text/src/lib/TextExplanationDashboard/Control/TextExplanationView/TextExplanationView.tsx

```diff
@@ -204,54 +213,21 @@ export class TextExplanationView extends React.PureComponent<
     this.setState({ singleTokenImportances });
   }
 
-  private calculateTopKImportances(importances: number[]): number {
-    return Math.min(
-      MaxImportantWords,
-      Math.ceil(Utils.countNonzeros(importances) / 2)
-    );
-  }
-
-  private calculateMaxKImportances(importances: number[]): number {
-    return Math.min(
-      MaxImportantWords,
-      Math.ceil(Utils.countNonzeros(importances))
-    );
+  private getImportanceForSingleToken(index: number): number[] {
+    return this.state.qaRadio === QAExplanationType.Start
+      ? this.props.dataSummary.localExplanations[0].map((row) => row[index])
+      : this.props.dataSummary.localExplanations[1].map((row) => row[index]);
   }
 
-  private computeImportancesForWeightVector(
-    importances: number[][],
-    weightVector: WeightVectorOption
-  ): number[] {
-    if (weightVector === WeightVectors.AbsAvg) {
-      // Sum the multidimensional array to one dimension across rows for each token
-      const numClasses = importances[0].length;
-      const sumImportances = importances.map((row) =>
-        row.reduce((a, b): number => {
-          return (a + Math.abs(b)) / numClasses;
-        }, 0)
-      );
-      return sumImportances;
+  private getBaseValue(): number {
+    if (this.props.dataSummary.baseValues) {
+      const retuVal =
```

Comment:
`retuVal` may be a redundant variable, and the expression can be directly returned as above